### PR TITLE
28_06_2024_tb_legacy_boot: add notes and results

### PR DESCRIPTION
**Pull request by macpijan**

### CODE REVIEW

File: notes/28_06_2024_tb_legacy_boot.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+3. It translate to following components:
+   - Landing Zone (LZ) - today known as SKL
+     - [recipe](https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/blob/49183efa4818987c7067ea4b35c3c25a9309b486/dynamic-layers/openembedded-layer/recipes-support/landing-zone/landing-zone_0.3.0.bb)
+     - [code revision](https://github.com/TrenchBoot/landing-zone/commit/4dc904bf70d6913a5706bdfc6f7781a9ff039d38)
+  - GRUB
+    - [recipe](https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/blob/49183efa4818987c7067ea4b35c3c25a9309b486/dynamic-layers/openembedded-layer/recipes-bsp/grub/grub-tb_git.bb)
+    - [code revision](https://github.com/3mdeb/grub2/tree/d6fa0d1f92446243e34e009361470f5730869ee9)
+  - Linux
+    - [recipe](https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/blob/49183efa4818987c7067ea4b35c3c25a9309b486/dynamic-layers/openembedded-layer/recipes-kernel/linux/linux-tb_5.5.bb)
+    - [code revision](https://github.com/TrenchBoot/linux/tree/eed5cdf480ee3761d18294d64ac7e2184229b51c)
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]: This lacks indentation to align with `Landing Zone` bullet.

File: notes/28_06_2024_tb_legacy_boot.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+## Building image
+
+1. Notes from building, although no need to do it - prebuild version
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]: Single-item list looks like something is missing.
```suggestion
Notes from building, although no need to do it - prebuild version
```

File: notes/28_06_2024_tb_legacy_boot.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+## Running image
+
+Image has been started on the PC Engines APU2 platform with
+[Dasharo (coreboot + SeaBIOS) v24.05.00.01](https://docs.dasharo.com/variants/pc_engines/releases_seabios/#v24050001-2024-06-28).
+
+## Test results
+
+### Booting
+
+Boot log can be available [here](../logs/28_06_2024_tb_legacy_boot.log).
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]:
```suggestion
Boot log is available [here](../logs/28_06_2024_tb_legacy_boot.log).
```

File: notes/28_06_2024_tb_legacy_boot.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+Result:
+
+```
+7251a94abb2ff0de0c2c6b89e16559e0e11605f4  SHA1
+a67681a54c9e46ae39316d7cb2c983d052f8190478b916224420dd5a719d358f  SHA256
+```
+
+2. Exeucted on the APU2:
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]:
```suggestion
2. Executed on the APU2:
```

File: notes/28_06_2024_tb_legacy_boot.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  17 : a67681a54c9e46ae39316d7cb2c983d052f8190478b916224420dd5a719d358f
+  18 : 7fd9068bb68527a096b44bf6831b8ad0cbb8b53e045bfd3ecc427bd68ec27de3
+  19 : 0000000000000000000000000000000000000000000000000000000000000000
+  20 : 0000000000000000000000000000000000000000000000000000000000000000
+  21 : 0000000000000000000000000000000000000000000000000000000000000000
+  22 : 0000000000000000000000000000000000000000000000000000000000000000
+  23 : 0000000000000000000000000000000000000000000000000000000000000000
+```
+
+SHA256 hash of PCR17 mathes to the pre-calculated one.
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]:
```suggestion
SHA256 hash of PCR17 matches to the pre-calculated one.
```

File: notes/28_06_2024_tb_legacy_boot.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+SHA1 does not.
+In the [blog post](https://blog.3mdeb.com/2020/2020-08-13-trenchboot-event-log/)
+both of them used to match.
```
Comment (SergiiDmytruk) [inline notes/28_06_2024_tb_legacy_boot.md]: Likely related to the use of a modern toolchain and https://github.com/TrenchBoot/secure-kernel-loader/commit/c00c554c793508a5719661105a6d6929249b0945.  IIRC, the issue manifested itself for SHA1 only.

**Review by SergiiDmytruk — COMMENTED** [review]
Typo/formatting nits and likely explanation of a mismatched PCR value.

File: notes/28_06_2024_tb_legacy_boot.md (hunk 6 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: Which toolchain? The one to build SKL/LZ?

Comment (SergiiDmytruk) [reply notes/28_06_2024_tb_legacy_boot.md]: Yes, although it's a yocto build, so maybe LZ was built manually for the test from the blog post.

Comment (krystian-hebel) [reply notes/28_06_2024_tb_legacy_boot.md]: Doesn't look like that is the issue. In the log, there is only one `shasum calculated:` and it looks like SHA256. This is how the code should work in https://github.com/TrenchBoot/landing-zone/blob/4dc904bf70d6913a5706bdfc6f7781a9ff039d38/main.c#L155, later it was improved to include SHA1 for TPM2 as well: https://github.com/TrenchBoot/landing-zone/blob/master/main.c#L162.

This was fixed on main branch on Jul 15, 2021, and blog post was released Aug 13, 2020. It was dormant on https://github.com/TrenchBoot/landing-zone/pull/52 before being rebased as https://github.com/TrenchBoot/landing-zone/pull/65 one year later. `meta-trenchboot` switched to our fork of LZ after the post (Nov 4, 2020 https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/commit/aa97eee114ed08c94bb71a4bf9efd75d85d43286), so it must've been manual build (less likely: time travel).

Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e - I have simply added some more context and linked this discussion as discussed with @krystian-hebel

File: notes/28_06_2024_tb_legacy_boot.md (hunk 5 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e

File: notes/28_06_2024_tb_legacy_boot.md (hunk 4 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e

File: notes/28_06_2024_tb_legacy_boot.md (hunk 3 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e

File: notes/28_06_2024_tb_legacy_boot.md (hunk 2 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e

File: notes/28_06_2024_tb_legacy_boot.md (hunk 1 above)
Comment (macpijan) [reply notes/28_06_2024_tb_legacy_boot.md]: 7febf807068f1fbb77dc14462f9696810998f16e

**Pull request merged**